kyverno-pss-policy-exceptions/README.md

```diff
@@ -0,0 +1,167 @@
+
+- access to exceptions can be controlled with Kubernetes-native RBAC
+- management of exceptions can be performed through GitOps workflows and eventually backed by other storage, like an OCI registry
+- Giant Swarm is able to create very flexible policy exceptions, for example, a sweeping emergency exception from a bad customer policy
+- additional optional controls about exceptions can be enforced by Kyverno itself, for example:
+  - requiring cryptographic signatures of `PolicyExceptions`
+  - limiting the scope of a `PolicyException`, as in these several [examples](https://kyverno.io/policies/other/policy-for-exceptions/policy-for-exceptions/)
+
+Under this model, whenever a resource requires an exception to a configured Kyverno policy, a `PolicyException` can be created by an end user, a security team, a platform team, Giant Swarm, or others, in accordance with cluster RBAC.
+
+This approach is very flexible, so we believe it will be palatable to customers with drastically different security cultures. More detailed examples are included below.
+
+### Giant Swarm Implementation
+
+Each Giant Swarm project currently contains a PSP.
+If the application requires permissions not allowed by our security policies, the application's PSP will be directly replaced by one or multiple `PolicyException` resources.
+If the application does not require any elevated permissions, it no longer requires any policy resources to be contained in the repository. The PSP will be removed and no replacement will be added.
+
+The first generation of Kyverno PSS-controlled Giant Swarm management clusters will be managed as described in the Developer-Set Exceptions example below. In short: each project deploys its own exceptions in management clusters and workload clusters.
+
+Future improvements to the security platform intend to centralize the exceptions and eliminate the need to maintain individual application-level resources, moving us to a Centralized Exception model.
+
+### Workload Cluster Default Implementation
+
+We currently include a default restricted PSP in each workload cluster. This will be replaced by Kyverno PSS policies, enforced in `restricted` mode.
+
+Customers will have great flexibility in how they deliver `PolicyExceptions` to the cluster for any workloads which are not compliant with the `restricted` standard. Examples of possible exception models are included below.
+
+A newly-created workload cluster will contain `PolicyException` resources deployed for our applications, described above.
+
+### Migration Implications
+
+If this approach is accepted, we expect the migration to involve the following effort from Giant Swarm and our customers.
+
+Giant Swarm:
+
+- creation of `PolicyExceptions` for our applications which require them and can't be made compliant
+  - Shield has already made many workloads compliant so they don't require exceptions. Others can still be made compliant.
```

Comment:
How?

Reply:
By fixing the violations in the Helm charts. For example, the `restricted` level requires workloads to explicitly `DROP` all capabilities. Most of our projects don't currently include this in the chart, but most don't require any extra capabilities, so the explicit drop can be (and is already being) added to the charts.

Reply:
maybe this is already planned, but can we have some docs for us and customers for these cases? with examples and such?

Reply:
Right, the first guide has already been published here: https://docs.giantswarm.io/advanced/security-policy-enforcement/